modules/tinymce/src/core/main/ts/caret/ClosestCaretCandidate.ts

```diff
@@ -0,0 +1,132 @@
+/**
+ * Copyright (c) Tiny Technologies, Inc. All rights reserved.
+ * Licensed under the LGPL or a commercial license.
+ * For LGPL see License.txt in the project root for license information.
+ * For commercial licenses see https://www.tiny.cloud/
+ */
+
+import { Arr, Optional } from '@ephox/katamari';
+import { Compare, SugarElement, Traverse } from '@ephox/sugar';
+
+import { getClientRects, NodeClientRect } from '../dom/Dimensions';
+import * as NodeType from '../dom/NodeType';
+import * as ClientRect from '../geom/ClientRect';
+import * as CaretCandidate from './CaretCandidate';
+import { isFakeCaretTarget } from './FakeCaret';
+
+type GeomClientRect = ClientRect.ClientRect;
+
+export enum FakeCaretPosition {
+  Before = 'before',
+  After = 'after'
+}
+
+export interface FakeCaretInfo {
+  node: Node;
+  position: FakeCaretPosition;
+}
+
+type distanceFn = <T extends GeomClientRect>(rect: T, x: number, y: number) => number;
```

Comment:
Can we please rename this to `DistanceFn`? as we never start types with a lower case character anywhere else in the code base and it looks too much like a variable.

Reply:
Is there a way we can lint this? ReScript is the opposite, types are _required_ to start with a lowercase letter. So it might come up a lot as some of us straddle the two languages.

Reply:
There is, however we currently have it disabled in this repo because we have a few cases of the opposite direction to fix or we need to come up with a rule to match the naming used in TinyMCE IIRC: https://github.com/tinymce/tinymce/blob/develop/.eslintrc.json#L12

Reply:
Yes, exactly my thought we probably need a eslint rule I will make this mistake again and maybe others that need to context switch between languages. I looked around for some built in option but couldn't find any might exist some plugin or we could write our own. Should be a fairly easy rule to write since it applies to all type declarations.

Reply:
Seems to be a rule that we might be able to use here https://github.com/typescript-eslint/typescript-eslint/blob/main/packages/eslint-plugin/docs/rules/naming-convention.md#enforce-the-codebase-follows-eslints-camelcase-conventions

Reply:
Added a PR here: https://github.com/tinymce/eslint-plugin/pull/27 but maybe we merge that after the Feb 15 release not super important just a nice to have but might break things if there is projects that are using lowercase.